e2e/nomostest/git-server.go

```diff
@@ -44,8 +44,8 @@ func testGitServerSelector() map[string]string {
 	return map[string]string{"app": testGitServer}
 }
 
-func setupGit(nt *NT) error {
-	if *e2e.GitProvider == e2e.Local {
+func setupSyncSources(nt *NT) error {
+	if *e2e.GitProvider == e2e.Local || *e2e.RegistryProvider == e2e.Local {
```

Comment:
why OR? Are the Git keys re-used for the local registry?

Reply:
Seems like maybe the CAs and PKs should be separate. We don't want to require that users use the same CA for both.

Reply:
Or do we only ever use one or the other and not both in any given test?

Reply:
I'm not sure I understand what you mean - could you elaborate?

Reply:
The only reason you made them share was to avoid duplicating the code, right? Not some technical constraint?

In most cases, the OCI registry and Git registry should be mutually exclusive for tests, right? But if we had some tests for migration they might not be.

It might make sense to just make one set for each to make it clear that they're not related and don't need to be the same. We wouldn't want to accidentally have some tests pass because they're the same, even tho nobody in the real world would have two services use the same private keys, even if they used the same root authority.

Reply:
> The only reason you made them share was to avoid duplicating the code, right? Not some technical constraint?

Yes, they share two things (just for simplicity/convenience and not a technical constraint):
- They live in the same Namespace
- They share a HTTPS certificate signed by the same root CA

> In most cases, the OCI registry and Git registry should be mutually exclusive for tests, right? But if we had some tests for migration they might not be.

I'm not sure what the distinction here is, but we do have tests that switch between git-sync and oci-sync (including the tests in this PR). Most tests just happen to focus mostly on one or the other.

> It might make sense to just make one set for each to make it clear that they're not related and don't need to be the same. We wouldn't want to accidentally have some tests pass because they're the same, even tho nobody in the real world would have two services use the same private keys, even if they used the same root authority.

Sure, I can generate different certs for each service from the same root CA. Although I'm struggling to imagine how this could cause false positives

Reply:
Risk reduction is about more than just avoiding predictable bugs. It's desirable to make the tests similar to how people are actually going to use the product for multiple reasons. The more realistic the tests, the more likely we are to catch real bugs, rather than synthetic bugs that customers will never see. Avoiding a shortcut now also helps prevent us from ignorantly using the code as a precedent in the future.

Of course, one alternative would be to exhaustively document the shortcut in enough places to be discoverable. Anyone who touches this in the future will need to know why it was done this way, for debugging, and to know why or why not to copy it.

Reply:
Done